lib/galaxy/webapps/galaxy/api/datasets.py

```diff
@@ -93,6 +93,52 @@ def show(self, trans, id, hda_ldda='hda', data_type=None, provider=None, **kwd):
             trans.response.status = 500
         return rval
 
+    @web._future_expose_api
+    def set_permissions(self, trans, dataset_id, payload, **kwargs):
+        """
+        POST /api/datasets/{encoded_dataset_id}/permissions
+        Updates permissions of a dataset.
+
+        :rtype:     dict
+        :returns:   dictionary containing new permissions
+        """
+        hda_ldda = kwargs.get('hda_ldda', 'hda')
+        try:
+            hda = self.get_hda_or_ldda(trans, hda_ldda=hda_ldda, dataset_id=dataset_id)
+        except Exception as e:
+            return str(e)
```

Comment:
This is not correct, better just to let the exception bubble up in an API controller and let the _future_expose_api framework exception handling code deal with it.

Reply:
Ah, ok, didn't know future_expose_api did that, awesome. That will save some boilerplate. Should I do that for *all* exceptions or just the ones where there is not some more informative exception message to return?